src/main/scala/com/databricks/spark/sql/perf/RunBenchmark.scala

```diff
@@ -101,39 +100,102 @@ object RunBenchmark {
+          avg($"optimizationTime") as 'avgOptimizationTimeMs,
+          stddev($"optimizationTime") as 'stdOptimizationDev,
+
+          min($"planningTime") as 'minPlanningTimeMs,
+          max($"planningTime") as 'maxPlanningTimeMs,
+          avg($"planningTime") as 'avgPlanningTimeMs,
+          stddev($"planningTime") as 'stdPlanningDev,
+
+          min($"executionTime") as 'minExecutionTimeMs,
+          max($"executionTime") as 'maxExecutionTimeMs,
+          avg($"executionTime") as 'avgExecutionTimeMs,
+          stddev($"executionTime") as 'stdExecutionDev)
         .orderBy("name")
-        .show(truncate = false)
-    println(s"""Results: sqlContext.read.json("${experiment.resultPath}")""")
-
-    config.baseline.foreach { baseTimestamp =>
-      val baselineTime = when($"timestamp" === baseTimestamp, $"executionTime").otherwise(null)
-      val thisRunTime = when($"timestamp" === experiment.timestamp, $"executionTime").otherwise(null)
-
-      val data = sqlContext.read.json(benchmark.resultsLocation)
-          .coalesce(1)
-          .where(s"timestamp IN ($baseTimestamp, ${experiment.timestamp})")
-          .withColumn("result", explode($"results"))
-          .select("timestamp", "result.*")
-          .groupBy("name")
-          .agg(
-            avg(baselineTime) as 'baselineTimeMs,
-            avg(thisRunTime) as 'thisRunTimeMs,
-            stddev(baselineTime) as 'stddev)
-          .withColumn(
-            "percentChange", ($"baselineTimeMs" - $"thisRunTimeMs") / $"baselineTimeMs" * 100)
-          .filter('thisRunTimeMs.isNotNull)
-
-      data.show(truncate = false)
-    }
+
+    runResults.show(runResults.count.toInt, truncate = false)
+
+    runResults
+      .coalesce(1)
+      .write
+      .option("header", "true")
+      .csv(experiment.csvResultPath)
+
+    println(s"""CSV Results: ${experiment.csvResultPath}""")
+  }
+
+  def compareResults(
+      sqlContext: SQLContext,
+      baselineTimestamp: Long,
+      targetTimestamp: Long,
+      resultsLocation: String,
+      outputLocation: String): Unit = {
+
+    import sqlContext.implicits._
+    val baselineTime = when($"timestamp" === baselineTimestamp, $"executionTime").otherwise(null)
+    val targetTime = when($"timestamp" === targetTimestamp, $"executionTime").otherwise(null)
+
+    val comparison = sqlContext.read.json(resultsLocation)
+        .coalesce(1)
+        .where(s"timestamp IN ($baselineTimestamp, $targetTimestamp)")
+        .withColumn("result", explode($"results"))
+        .select("timestamp", "result.*")
+        .groupBy("name")
+        .agg(
+          avg(baselineTime) as 'baselineTimeMs,
+          avg(targetTime) as 'targetTimeMs,
+          stddev(baselineTime) as 'stddev)
+        .withColumn(
+          "percentChange", ($"baselineTimeMs" - $"targetTimeMs") / $"baselineTimeMs" * 100)
+        .filter('targetTimeMs.isNotNull)
+
+    comparison.show(comparison.count.toInt, truncate = false)
```

Comment:
nit: don't want some order by (e.g. by query name)  before showing / writing it?